plugins/processors/starlark/starlark_test.go

```diff
@@ -3256,3 +3279,22 @@ func parseErrorMessage(t *testing.T, lines []string, header string) string {
 	require.True(t, startIdx < len(lines), fmt.Sprintf("Expected to find the error message after %q, but found none", header))
 	return strings.TrimLeft(lines[startIdx], "# ")
 }
+
+func testLoadFunc(module string, logger telegraf.Logger) (starlark.StringDict, error) {
+	result, err := loadFunc(module, logger)
+	if err != nil {
+		return nil, err
+	}
+
+	if module == "time.star" {
+		customModule := result["time"].(*starlarkstruct.Module)
+		customModule.Members["now"] = starlark.NewBuiltin("now", testNow)
+		result["time"] = customModule
+	}
+
+	return result, nil
+}
+
```

Comment:
IIRC you can change the `NowFunc` of the time module at runtime after initialization. Have you tried this?

Reply:
That possibly could work (would have to ensure its set before loadFunc is called) but IMO overriding the method for tests allows other tests to re-use the same 'test' load where we can override functions that otherwise couldn't be tested, like time.now.

Reply:
Well it means that you add new modules in two places now, but it seems you made your choice already anyway...